openvino_wrapper_lib/src/models/base_model.cpp

```diff
@@ -89,3 +90,54 @@ Models::ObjectDetectionModel::ObjectDetectionModel(
   const std::string& model_loc,
   int max_batch_size)
 : BaseModel(label_loc, model_loc, max_batch_size) {}
+
+cv::Mat Models::ObjectDetectionModel::resizeImage(const cv::Mat &orig_image, const int resized_width, const int resized_height,
+  double fx, double fy, int interpolation, int border_dt, int border_db, int border_dl, int border_dr, cv::Scalar border_color) {
+    cv::Mat resized_image;
+    if (resized_width != orig_image.size().width || resized_height != orig_image.size().height) {
+      cv::resize(orig_image, resized_image, cv::Size(resized_width, resized_height), fx, fy, interpolation);
+    }
+    if ( border_dt != 0 || border_db != 0 || border_dl != 0 || border_dr != 0) {
+      cv::copyMakeBorder(resized_image,
+                         resized_image,
+                         border_dt,
+                         border_db,
+                         border_dl,
+                         border_dr,
+                         cv::BORDER_CONSTANT,
+                         border_color);
+    }
+    return resized_image;
+}
+
+void Models::ObjectDetectionModel::dataToBlob(cv::Mat& resize_img, float scale_factor, int batch_index,
+  const std::shared_ptr<Engines::Engine> & engine, bool mat_steps) {
+    std::string input_name = getInputName();
+    slog::debug << "add input image to blob: " << input_name << slog::endl;
+    ov::Tensor in_tensor = engine->getRequest().get_tensor(input_name);
+    u_int8_t * blob_data = (u_int8_t *)in_tensor.data();
+
+    size_t height = resize_img.rows;
+    size_t width = resize_img.cols;
+    auto *input_data = (float *) resize_img.data;
+
+    ov::Shape in_shape = in_tensor.get_shape();
+    long unsigned int channels;
+    if (mat_steps) {
+      channels = in_shape[1];
+      int batchOffset = batch_index * width * height * channels;
+      for (int c = 0; c < channels; c++) {
+        for (int h = 0; h < height; h++) {
+          for (int w = 0; w < width; w++) {
+            blob_data[batchOffset + c * width * height + h * width + w] =
+              resize_img.at<cv::Vec3b>(h, w)[c] * scale_factor;
+          }
+        }
+      }
+    } else {
+      channels = in_shape[3];
```

Comment:
You set some assumptions here, which is not good:

- The model's input layout is NHWC

Can we have some optimizaiton on this?